sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -226,6 +226,9 @@ class JDBCOptions(
   // The prefix that is added to the query sent to the JDBC database.
   // This is required to support some complex queries with some JDBC databases.
   val prepareQuery = parameters.get(JDBC_PREPARE_QUERY).map(_ + " ").getOrElse("")
+
+  // Infers timestamp values as TimestampNTZ type when reading data.
+  val inferTimestampNTZType = parameters.getOrElse(JDBC_INFER_TIMESTAMP_NTZ, "false").toBoolean
```

Comment:
Should we maybe check if `spark.sql.timestampType` is `TIMESTAMP_NTZ` if `inferTimestampNTZType` is not set? That's how CSV type inference and Python type inference do.

Reply:
cc @gengliangwang FYI

Reply:
Yes, I thought about it, let's ask @gengliangwang.